biz.aQute.bndlib/src/aQute/bnd/osgi/PluginsContainer.java

```diff
@@ -378,6 +382,12 @@ private Object loadPlugin(Processor processor, ClassLoader loader, Attrs attrs,
 		return null;
 	}
 
+	public void addCloseable(Object plugin) {
```

Comment:
This breaks the thread safety. `plugins` is a copy-on-write set, so it is safe to add plugins after the container is initialized.  But `closeablePlugins` is a plain HashSet.

Reply:
https://github.com/bndtools/bnd/blob/ab6abc8987ebc7f33b80c10c6dc05fc476e3ee61/biz.aQute.bndlib/src/aQute/bnd/osgi/PluginsContainer.java#L50-L53